app/controllers/sites_controller.rb

```diff
@@ -1,5 +1,17 @@
 class SitesController < ApplicationController
   before_action :set_site
-  load_and_authorize_resource
+  load_and_authorize_resource instance_variable: :site, class: 'Site' # descendents still auth Site
   layout 'dashboard'
+
+  def update
```

Comment:
Yeah, let's hold up my actual improvement for your hypothetical problem... again.  Not feeling it.  You can file an issue to refactor and we'll let it rot on the backlog like we do with mine.